src/lib/bmp/bmp.c

```diff
@@ -0,0 +1,194 @@
+/**
+ * @file
+ * @brief lib for read and draw BMP
+ *
+ * @date Apr 18, 2019
+ * @author Filipp Chubukov
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/ioctl.h>
+
+#include <linux/fb.h>
+
+/* now it works with non pallete bmp formats - 24 and 32 bits, tested on 32bpp fb and not tested on 16bpp fb */
+typedef struct {
+	unsigned short bfType;	      /* == "BM" */
+	unsigned int bfSize;	      /* Size of file */
+	unsigned int bfReserved;      /* == "00" */ 
+	unsigned int bfOffBits;	      /* Indicate to start of image bytes */
+	unsigned int biSize;	      /* Size of header */
+	int biWidth;		      /* Width of image in pixels */
+	int biHeight;		      /*Height of image in pixels */
+	unsigned int biPlanes;	      /* == 1 */
+	int biBitCount;		      /* bits per pixel */
+	int biCompression;	      /* type of compression(we are not using) */
+	unsigned int biSizeImage;     /* size of image, == 0 if no compression */
+	unsigned int biXPelsPerMeter; /* X resolution */
+	unsigned int biYPelsPerMeter; /* Y resolution */
+	unsigned int biClrUsed;	      /* Number of used from color table colors */
+	unsigned int biClrImportant;  /* Number of important colors - min for drawing */ 
+} bmp_header;
+
+int read_bmp_head(int fd ,bmp_header *bmp_head) { /* reading bmp header */
+	read(fd,&(bmp_head->bfType), 2);
```

Comment:
Why not just `read(fd, bmp_head, sizeof(bmp_header))`?

Reply:
your example doesn't work and I don't see a problem with mine

Reply:
Lots of lines of code which are almost the same is a problem.

As far as I understand, it just reads sequential fields from the file (i.e. structure is the same as header in the file), so reading whole header in one system call should work.

Reply:
done, remove this fun, also add new fun for predict size of image, add some error checks and little changes